mavisp/modules.py

```diff
@@ -105,76 +105,69 @@ def ingest(self, mutations):
-                this_error = f"Only using one single instance of any given method is supported"
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-
-        for model in models:
-
-            method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
-
-            for method_dir in method_dirs:
-                self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
-                self.data = self.data.join(self.methods[method_dir].data)
-
-        keys = [ k for k in self.data.keys() if k.startswith('Stability') ]
-
-        # check if we have FoldX column
-        foldx_col = [k for k in keys if 'FoldX' in k]
-        if len(foldx_col) == 1:
-            foldx_header = foldx_col[0]
-        else:
-            foldx_header = None
-
-        if any(['Rosetta' in k for k in keys]):
-            rosetta_col = [k for k in keys if 'Rosetta' in k]
-            assert len(rosetta_col) == 1
-            rosetta_header = rosetta_col[0]
-        else:
-            rosetta_header = None
+            for model in models:
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                if not set(method_dirs).issubset(set(self.methods.keys())):
+                    this_error = f"One or more {self.name} methods are not supported"
+                    raise MAVISpMultipleError(warning=warnings,
+                                            critical=[MAVISpCriticalError(this_error)])
+                all_methods.extend(method_dirs)
+                if len(all_methods) != len(set(all_methods)):
+                                this_error = f"Only using one single instance of any given method is supported"
+                                raise MAVISpMultipleError(warning=warnings,
+                                                        critical=[MAVISpCriticalError(this_error)])
+            for model in models:
+
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                for method_dir in method_dirs:
+                    parsed_data = self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
+                    self.data = self.data.join(parsed_data, rsuffix= " using " + f'{method}' + " method")
+
+            keys = [ k for k in self.data.keys() if k.startswith('Stability') ]
+
+            # check if we have FoldX column
+            if any(['FoldX' in k for k in keys]):
+                foldx_cols = [k for k in keys if 'FoldX' in k]
+                for foldx_col in foldx_cols:
+                    foldx_header = foldx_col
```

Comment:
I'm not sure what you're trying to accomplish here - all this does is to set `foldx_header`to the last element of the `foldx_col` list. It is functionally equivalent to `foldx_header = foldx_cols[-1]`. for instance:

```
In [1]: l = [1, 2, 3, 4]

In [2]: for i in l:
   ...:     this_l = i
   ...:

In [3]: print(this_l)
4
```